File: app/src/googlePlay/java/com/genonbeta/TrebleShot/activity/DonationActivity.java
package com.genonbeta.TrebleShot.activity;

import android.os.Bundle;
import android.view.MenuItem;
import android.view.animation.AlphaAnimation;
import android.view.animation.Animation;
import android.widget.TextView;

import com.genonbeta.TrebleShot.R;
import com.genonbeta.TrebleShot.app.Activity;
import com.genonbeta.TrebleShot.fragment.external.InAppDonationItemListFragment;

import androidx.annotation.Nullable;
import androidx.appcompat.widget.Toolbar;

/**
 * created by: veli
 * date: 7/12/18 10:32 PM
 */
public class DonationActivity extends Activity
{
	private Animation mAnimation;
	private TextView mDeveloperText;
	private String[] mTexts;

	@Override
	protected void onCreate(@Nullable Bundle savedInstanceState)
	{
		super.onCreate(savedInstanceState);
		setContentView(R.layout.activity_donation);

		Toolbar toolbar = findViewById(R.id.toolbar);
		setSupportActionBar(toolbar);

		mDeveloperText = findViewById(R.id.developerText);
		mTexts = getString(R.string.text_prefaceDonation).split(";");

		if (getSupportActionBar() != null)
			getSupportActionBar().setDisplayHomeAsUpEnabled(true);

		InAppDonationItemListFragment contributorsListFragment = (InAppDonationItemListFragment) getSupportFragmentManager().findFragmentById(R.id.activity_about_contributors_fragment);

		if (contributorsListFragment != null)
			contributorsListFragment.getListView().setNestedScrollingEnabled(false);

		AlphaAnimation alphaAnimation = new AlphaAnimation(0.0f, 1.0f);

		alphaAnimation.setDuration(3000);
		alphaAnimation.setRepeatCount(Animation.INFINITE);
		alphaAnimation.setRepeatMode(Animation.REVERSE);
		alphaAnimation.setAnimationListener(new Animation.AnimationListener()
		{
			int mAnimPoint;
			int mTextPoint;

			@Override
			public void onAnimationStart(Animation animation)
			{
				mAnimPoint = 0;
				mTextPoint = 0;

				mDeveloperText.setText(mTexts[mTextPoint]);
				changeSpeed();
			}

			@Override
			public void onAnimationEnd(Animation animation)
			{

			}

			@Override
			public void onAnimationRepeat(Animation animation)
			{
				mAnimPoint++;

				if (mAnimPoint % 2 == 0) {
					mAnimPoint = 0;
					mTextPoint = mTextPoint + 1 >= mTexts.length ? 0 : mTextPoint + 1;
					mDeveloperText.setText(mTexts[mTextPoint]);

					changeSpeed();
				}
			}

			void changeSpeed() {
				mAnimation.setDuration(30 * mDeveloperText.getText().length());
			}
		});

		mAnimation = alphaAnimation;
	}

	@Override
	protected void onResume()
	{
		super.onResume();
		mDeveloperText.setAnimation(mAnimation);
	}

	@Override
	protected void onPause()
	{
		super.onPause();
		mDeveloperText.setAnimation(null);
	}

	@Override
	public boolean onOptionsItemSelected(MenuItem item)
	{
		int id = item.getItemId();

		if (id == android.R.id.home)
			onBackPressed();
		else
			return super.onOptionsItemSelected(item);

		return true;
	}
}

File: app/src/googlePlay/java/org/monora/uprotocol/client/android/activity/DonationActivity.kt
package org.monora.uprotocol.client.android.activity

import android.os.Bundle
import android.view.animation.AlphaAnimation
import android.view.animation.Animation
import android.widget.TextView
import org.monora.uprotocol.client.android.R
import org.monora.uprotocol.client.android.app.Activity
import org.monora.uprotocol.client.android.fragment.external.DonationListFragment

/**
 * created by: veli
 * date: 7/12/18 10:32 PM
 */
class DonationActivity : Activity() {
    private val animation: Animation by lazy {
        AlphaAnimation(0.0f, 1.0f).also {
            it.duration = 3000
            it.repeatCount = Animation.INFINITE
            it.repeatMode = Animation.REVERSE

            it.setAnimationListener(object : Animation.AnimationListener {
                var animPoint = 0

                var textPoint = 0

                override fun onAnimationStart(animation: Animation) {
                    animPoint = 0
                    textPoint = 0
                    developerText.text = texts[textPoint]

                    changeSpeed()
                }

                override fun onAnimationEnd(animation: Animation) {

                }

                override fun onAnimationRepeat(animation: Animation) {
                    animPoint++

                    if (animPoint % 2 == 0) {
                        animPoint = 0
                        textPoint = if (textPoint + 1 >= texts.size) 0 else textPoint + 1
                        developerText.text = texts[textPoint]

                        changeSpeed()
                    }
                }

                fun changeSpeed() {
                    animation.duration = (30 * developerText.text.length).toLong()
                }
            })
        }
    }

    private val developerText: TextView by lazy {
        findViewById(R.id.developerText)
    }

    private val texts: Array<String> by lazy {
        getString(R.string.text_prefaceDonation).split(";").toTypedArray()
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_donation)
        setSupportActionBar(findViewById(R.id.toolbar))
        supportActionBar?.setDisplayHomeAsUpEnabled(true)

        val contributorsListFragment = supportFragmentManager.findFragmentById(
            R.id.activity_about_contributors_fragment
        ) as DonationListFragment?

        contributorsListFragment?.listView?.isNestedScrollingEnabled = false
    }

    override fun onResume() {
        super.onResume()
        developerText.animation = animation
    }

    override fun onPause() {
        super.onPause()
        developerText.animation = null
    }
}